### PR TITLE
Factorize more guiEditBoxes code

### PR DESCRIPTION
**Pull request by nerzhul**
Add compact, short information about your PR for easier understanding:

- Reduce duplicated code between editBoxes in our code

## To do

This PR is Ready for Review.

## How to test

Run code with various editboxes, all should work as intended

### CODE REVIEW

File: src/gui/guiEditBox.cpp [hunk 1]
```diff
@@ -689,6 +689,49 @@ bool GUIEditBox::onKeyDelete(const SEvent &event, s32 &mark_begin, s32 &mark_end
...
+
+			if (m_mark_begin != m_mark_end) {
+				// replace marked text
+				const s32 realmbgn = m_mark_begin < m_mark_end
+								     ? m_mark_begin
+								     : m_mark_end;
+				const s32 realmend = m_mark_begin < m_mark_end
+								     ? m_mark_end
+								     : m_mark_begin;
+
```
Comment (SmallJoker) [inline src/gui/guiEditBox.cpp:709]: Bad style

File: src/gui/guiEditBox.cpp [hunk 2]
```diff
@@ -817,3 +860,54 @@ void GUIEditBox::updateVScrollBar()
...
+	out->addBool("PasswordBox", m_passwordbox);
+	core::stringw ch = L" ";
+	ch[0] = m_passwordchar;
+	out->addString("PasswordChar", ch.c_str());
+	out->addEnum("HTextAlign", m_halign, GUIAlignmentNames);
+	out->addEnum("VTextAlign", m_valign, GUIAlignmentNames);
+	out->addBool("Writable", m_writable);
+
+	IGUIEditBox::serializeAttributes(out, options);
+}
```
Comment (SmallJoker) [inline src/gui/guiEditBox.cpp:913]: \+ \\n

**Review by SmallJoker — COMMENTED** [review]
Works

File: src/gui/guiEditBox.cpp (hunk 1 above)
Comment (nerzhul) [reply src/gui/guiEditBox.cpp]: fixed

File: src/gui/guiEditBox.cpp (hunk 2 above)
Comment (nerzhul) [reply src/gui/guiEditBox.cpp]: fixed

**Pull request merged**